### PR TITLE
Refactor _done of Watching in compiler

### PR DESCRIPTION
**Pull request by timse**
based off of #3933 but could be made independent.

**What kind of change does this PR introduce?**

refactoring

**Did you add tests for your changes?**

No

**If relevant, link to documentation update:**

N/A

**Summary**

Simplify the _done method of Watching in Compiler

**Does this PR introduce a breaking change?**

No

### CODE REVIEW

File: lib/Compilation.js [hunk 1]
```diff
@@ -1082,10 +1081,6 @@ class Compilation extends Tapable {
 		return this.mainTemplate.applyPluginsWaterfall("asset-path", filename, data);
 	}
 
-	getStats() {
```
Comment (sokra) [inline lib/Compilation.js]: That's a breaking change for no valid reason

**Comment by timse** [comment]
happy to give the handler a `Stats` Object and just mark a todo to maybe remove that in a later version?
That way we can get rid off this PR :).

**Comment by sokra** [comment]
Thanks

File: lib/Compiler.js [hunk 2]
```diff
@@ -79,22 +77,29 @@ Watching.prototype._go = function() {
...
 Watching.prototype._done = function(err, compilation) {
 	this.running = false;
 	if(this.invalid) return this._go();
-	this.error = err || null;
-	this.stats = compilation ? compilation.getStats() : null;
-	if(this.stats) {
-		this.stats.startTime = this.startTime;
-		this.stats.endTime = new Date().getTime();
+
+	var stats = this._getStats(compilation);
```
Comment (satazor) [inline lib/Compiler.js]: This line throws an error if there's no compilation.. e.g.: if `watch-run` plugin handlers call callback with an error, compilation will be null.

@sokra @timse

**Pull request merged**